### PR TITLE
Add verbose debug logging to biosdisk and gpt

### PR DESCRIPTION
**Pull request by marineam**
The biosdisk driver now logs extensively about what it is doing. The information provided is similar to efidisk's logging. Will post this one upstream after merging.

My gpt module similarly didn't say anything about what it was doing, fixed now. :)

### CODE REVIEW

**Comment by marineam** [comment]
Made a couple minor tweaks and rebased.

File: grub-core/lib/gpt.c [hunk 1]
```diff
@@ -237,6 +249,64 @@ grub_gpt_header_check (struct grub_gpt_header *gpt,
 }
 
 static grub_err_t
+grub_gpt_check_primary (grub_gpt_t gpt)
```
Comment (vcaputo) [inline grub-core/lib/gpt.c]: Everything looks fine, except I'm curious why you didn't just add logged variant of `grub_gpt_sector_to_addr` with the same signature just with the addition of a context string for "primary" vs. "secondary" to stick in the logs.  The current code seems a bit of unnecessarily copy-and-paste

Comment (marineam) [inline grub-core/lib/gpt.c]: The two new check_backup and check_primary functions make more sense in the next PR after this one where I add additional tests that are specific to primary and backup headers and therefore cannot be shared in the header_check function.

Comment (vcaputo) [inline grub-core/lib/gpt.c]: Ah, ok.

**Comment by vcaputo** [comment]
lgtm

**Pull request merged**